framework/src/consumption/lib/redshift-serverless/redshift-serverless-namespace.ts

```diff
@@ -0,0 +1,185 @@
+
+  /**
+   * The name of the created namespace
+   */
+  readonly namespaceName: string;
+
+  /**
+   * The created Secrets Manager secret containing the admin credentials
+   */
+  readonly adminSecret: ISecret;
+
+  /**
+   * The roles attached to the namespace
+   */
+  readonly roles: Record<string, IRole>;
+
+  /**
+   * KMS key used by the namespace to encrypt its data
+   */
+  readonly namespaceKey: Key;
+
+  /**
+   * The name of the database
+   */
+  readonly dbName: string;
+
+  /**
+   * The ARN of the created namespace
+   */
+  readonly namespaceArn: string;
+
+  /**
+   * The ID of the created namespace
+   */
+  readonly namespaceId: string;
+
+  private readonly removalPolicy: RemovalPolicy;
+  private readonly currentStack: Stack;
+  private namespaceParameters: { [key:string]: any };
+
+  constructor(scope: Construct, id: string, props: RedshiftServerlessNamespaceProps) {
+    const trackedConstructProps: TrackedConstructProps = {
+      trackingTag: RedshiftServerlessNamespace.name,
+    };
+
+    super(scope, id, trackedConstructProps);
+
+    this.roles = {};
+    this.currentStack = Stack.of(this);
+
+    if (props.iamRoles) {
+      for (const role of props.iamRoles) {
+        this.roles[role.roleArn] = role;
+      }
+    }
+
+    if (props.defaultIAMRole) {
+      this.roles[props.defaultIAMRole.roleArn] = props.defaultIAMRole;
+    }
+
+    this.dbName = props.dbName;
+    this.removalPolicy = Context.revertRemovalPolicy(scope, props.removalPolicy);
+    const logExports: string[] = props.logExports || [];
+    this.namespaceName = `${props.name}-${Names.uniqueResourceName(this, { maxLength: 64 - (props.name.length + 1) }).toLowerCase()}`;
+
+    this.namespaceKey = props.kmsKey || new Key(this, 'DefaultNamespaceKey', { enableKeyRotation: true });
+    const namespaceArn = `arn:aws:redshift-serverless:${this.currentStack.region}:${this.currentStack.account}:namespace/*`;
+
+    this.namespaceParameters = {
+      namespaceName: this.namespaceName,
+      adminPasswordSecretKmsKeyId: this.namespaceKey.keyId,
+      adminUsername: `admin-${randomBytes(4).toString('hex')}`,
+      dbName: props.dbName,
+      defaultIamRoleArn: props.defaultIAMRole ? props.defaultIAMRole.roleArn : undefined,
+      iamRoles: this.roles ? Object.keys(this.roles) : undefined,
+      kmsKeyId: this.namespaceKey.keyId,
+      manageAdminPassword: true,
+      logExports,
+    };
+
+    const roleArns = Object.keys(this.roles);
+
+    const createNamespaceCrPolicyStatements = [
+      new PolicyStatement({
+        effect: Effect.ALLOW,
+        actions: [
+          'redshift-serverless:CreateNamespace',
+          'redshift-serverless:UpdateNamespace',
+          'redshift-serverless:DeleteNamespace',
+        ],
+        resources: [namespaceArn],
+      }),
+      new PolicyStatement({
+        effect: Effect.ALLOW,
+        actions: [
+          'secretsmanager:CreateSecret',
+          'secretsmanager:RotateSecret',
```

Comment:
can this statement of rotate be isolated to only the secrets we manage?
Same for Tag can we restrict to work only with the tags DSF is managing?

Reply:
It's a service owned secret, I can check if we can isolate it via tags. The API call to create namespace is the one that also creates the secret.